flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestServerEndpoint.java

```diff
@@ -147,40 +170,65 @@ public InetSocketAddress getServerAddress() {
 	}
 
 	/**
+	 * Returns the address of the REST server endpoint. Since the address is only known
+	 * after the endpoint is started, it is returned as a future which is completed
+	 * with the REST address at start up.
+	 *
+	 * @return REST address of this endpoint
+	 */
+	public String getRestAddress() {
```

Comment:
What is the relation of this method to `getServerAddress()`?

Reply:
This will return the fully qualified URL including protocol whereas `getServerAddress` only return the address of the server to which it is bound.